src/elements/content-explorer/ContentExplorer.js

```diff
@@ -1275,33 +1275,23 @@ class ContentExplorer extends Component<Props, State> {
      * @param {Object} item file or folder
      * @returns {void}
      */
-    handleSharedLinkSuccess = (newItem: BoxItem) => {
+    handleSharedLinkSuccess = (item: BoxItem) => {
         const { currentCollection } = this.state;
 
-        if (!newItem[FIELD_SHARED_LINK]) {
-            const { canSetShareAccess }: Props = this.props;
-            if (!newItem || !canSetShareAccess) {
-                return;
-            }
-
-            const { permissions, type } = newItem;
-            if (!permissions || !type) {
-                return;
-            }
```

Comment:
couple things here:

1. we don't want to `return` here because the collection should still be updated
2. the `if (!newItem)`check is unnecessary since we already check `!newItem[FIELD_SHARED_LINK]` in the outer block
3. the `canSetShareAccess` prop should not affect whether a user can generate a share link, only whether the user can update the access level
    -  generating a share link is checked with the `canShare` prop which is handled in `shareCallback()`

Reply:
For 3 I think the naming can be more precise, something like `canGenerateSharedLink` rather than `canShare`, and `canUpdateSharedLinkAccess` or even `canUpdateShareAccess` rather than `canSetShareAccess`. And also `shareCallback` probably has a better name too

Reply:
depending on the names you're referring to, it's probably not possible to change them:

- `canShare` is a prop that controls rendering the "Share" button
- `canSetShareAccess` is a prop that controls rendering a `select` element that allows the user to select an access level: [reference](https://user-images.githubusercontent.com/1075325/27887156-0940ee3e-6194-11e7-8e22-961139e82dfe.png)
- `can_share` (as a variable: `canShare`) and `can_set_share_access` (as a variable: `canSetShareAccess`) refer to item permissions from the backend

for the props, we can't change these names without a breaking change since developers pass these into the elements. and it's not worth changing the backend permissions

the `shareCallback` follows the pattern of the other callbacks: `renameCallback`, `deleteCallback`, `previewCallback`. not great names but consistent 🤷 

we can change the other stand alone variables though (like `canSkipSetShareAccessPermission`) but were there any instances in particular you were thinking about?